webapp/channels/src/components/integrations/abstract_oauth_app.tsx

```diff
@@ -455,14 +501,11 @@ export default class AbstractOAuthApp extends React.PureComponent {
                                 className='btn btn-primary'
                                 type='submit'
                                 spinning={this.state.saving}
-                                spinningText={localizeMessage(this.props.loading.id, this.props.loading.defaultMessage)}
+                                spinningText={typeof this.props.loading === 'string' ? this.props.loading : Utils.localizeMessage(this.props.loading?.id ?? '', this.props.loading?.defaultMessage as string)}
```

Comment:
If we can keep the `loading` type as `MessageDescriptor` only, we dont need this change